aiverify-foundation/aiverify
# Add validation of widgets, input blocks and templates during plugin scans

## Description

Changes:
- add `aiverify-shared-library` project.
- update stock widgets and input blocks to use new `aiverify-shared-library` module name.
- add image samples under `stock-plugins/aiverify.stock.robustness-toolbox/widgets/images/` to fix widget import errors.
-  add new sub-project `aiverify-apigw-node` which contains validation node scripts.
- update plugin scans to include validation and reading of widgets, input blocks and templates.
- Save MDX bundles files under file store.
- Add new pydantic schemas and db models for widgets, input blocks and templates
- add new endpoints under `plugin_router` to allow user to download widgets and input blocks zips and MDX bundles. 

## Motivation and Context

[Explain the motivation or the context behind this pull request. Why is it necessary?]

## Type of Change












feat: Add validation of widgets, input blocks and templates during plugin scans

## How to Test

**Unit Testing**:
Run `hatch test` to run the unit tests under `aiverify-apigw`.

**Manual Testing**:
If existing plugin tables exist in the SQL db, delete the tables to trigger re-scan of stock plugins.
Run `hatch run dev` to run the apigw. Navigate to http://localhost:4000/docs to view the API documentation.
Under plugin route, test:
- GET "/plugins" and check that all stock plugins data return include, widgets, input blocks and template data.
- GET "/plugins/{gid}/widgets" and check that the widgets zip is returned for the plugin.
- GET "/plugins/{gid}/input_blocks" and check that the input blocks zip is returned for the plugin.
- GET "/plugins/{gid}/bundle/{cid}" and check that the MDX bundle for widget or input block is returned.
- GET "/plugins/{gid}/summary/{cid}" and check that the MDX bundle input block summary is returned.

## Checklist

Please check all the boxes that apply to this pull request using "x":


## Screenshots (if applicable)

![image](https://github.com/user-attachments/assets/3f8dd962-4861-4619-aba9-ee50c8c47f73)

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
